.github/workflows/core_build.yml

```diff
@@ -0,0 +1,59 @@
+name: core-build
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        compiler: [clang6, clang7]
+        modules: [with, without]
+    runs-on: ubuntu-18.04
+    name: ${{ matrix.compiler }}-${{ matrix.modules }}-modules
+    env:
+      COMPILER: ${{ matrix.compiler }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Checkout modules
+        run: ./apps/ci/ci-install-modules.sh
+        if: matrix.modules == 'with'
```

Comment:
shouldn't the `if` come before `run`, or this order is mandatory?

Reply:
The order does not matter.
Its basically an object with attributes (name, run, if) that we define here and all values will be available when it is used.
Can change the if to be before if that is clearer in some way.